src/compiler/transformers/esDecorators.ts

```diff
@@ -0,0 +1,2087 @@
+                //     ...
+
+                switch (node.operatorToken.kind) {
+                    case SyntaxKind.EqualsToken:
+                    case SyntaxKind.AmpersandAmpersandEqualsToken:
+                    case SyntaxKind.BarBarEqualsToken:
+                    case SyntaxKind.QuestionQuestionEqualsToken:
+                        if (isIdentifier(node.left) &&
+                            isDecoratedAnonymousClassExpression(node.right)) {
+                            const assignedName = getAssignedNameOfIdentifier(node.left, node.right);
+                            const left = visitNode(node.left, visitor, isExpression);
+                            const right = visitNode(node.right, node => namedEvaluationVisitor(node, assignedName), isExpression);
+                            return factory.updateBinaryExpression(node, left, node.operatorToken, right);
+                        }
+                        break;
+                }
+
+                if (isSuperProperty(node.left) && classThis && classSuper) {
+                    let setterName =
+                        isElementAccessExpression(node.left) ? visitNode(node.left.argumentExpression, visitor, isExpression) :
+                        isIdentifier(node.left.name) ? factory.createStringLiteralFromNode(node.left.name) :
+                        undefined;
+                    if (setterName) {
+                        // super.x = ...
+                        // super.x += ...
+                        // super[x] = ...
+                        // super[x] += ...
+                        let expression = visitNode(node.right, visitor, isExpression);
+                        if (isCompoundAssignment(node.operatorToken.kind)) {
+                            let getterName = setterName;
+                            if (!isSimpleInlineableExpression(setterName)) {
+                                getterName = factory.createTempVariable(hoistVariableDeclaration);
+                                setterName = factory.createAssignment(getterName, setterName);
+                            }
+
+                            const superPropertyGet = factory.createReflectGetCall(
+                                classSuper,
+                                getterName,
+                                classThis);
+                            setOriginalNode(superPropertyGet, node.left);
+                            setTextRange(superPropertyGet, node.left);
+
+                            expression = factory.createBinaryExpression(
+                                superPropertyGet,
+                                getNonAssignmentOperatorForCompoundAssignment(node.operatorToken.kind),
+                                expression
+                            );
+                            setTextRange(expression, node);
+                        }
+
+                        const temp = discarded ? undefined : factory.createTempVariable(hoistVariableDeclaration);

```

Comment:
it would helpful to have an code example comment here

Reply:
I'll add a comment. This basically just avoids emitting unused temp variables for `super.x++`.